### PR TITLE
Feature: FavoriteAdCell

### PR DESCRIPTION
**Pull request by bstien**
# Why?
We need a new cell type for our redesign. This will be used when listing favorited ads within a folder.

The icon for the "more" button is probably not the correct one, so I'll keep that in mind until Thuy is back from 🏖🍹

I'm not sure of the access level set on the design spec and won't post it here, so just ask if you want it🤓

# What?
- Added cell `FavoriteAdTableViewCell`.
- Make it possible to set a `RibbonView`'s style after creating it.
- Added demo + snapshot reference images.

# Show me
Screenshot 2019-07-30 at 14 04 38

### CODE REVIEW

File: Demo/Cells/FavoriteAdTableViewCell/FavoriteAdCellDemoView.swift [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+}
+
+// MARK: - UITableViewDelegate
+
+extension FavoriteAdCellDemoView: UITableViewDelegate {
+    func tableView(_ tableView: UITableView, willDisplay cell: UITableViewCell, forRowAt indexPath: IndexPath) {
+        let isLastCell = indexPath.row == (viewModels.count - 1)
```
Comment (robinsalehjan) [inline Demo/Cells/FavoriteAdTableViewCell/FavoriteAdCellDemoView.swift]: Nice one 👍

File: Sources/Cells/FavoriteAdTableViewCell/FavoriteAdTableViewCell.swift [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+        let button = UIButton(withAutoLayout: true)
+        let image = UIImage(named: .more).withRenderingMode(.alwaysTemplate)
+        button.setImage(image, for: .normal)
+        button.imageView?.contentMode = .scaleAspectFit
+        button.tintColor = .stone
+        button.addTarget(self, action: #selector(moreButtonTapped), for: .touchUpInside)
+        return button
+    }()
+
+    private lazy var statusRibbon: RibbonView = {
```
Comment (vadymmarkov) [inline Sources/Cells/FavoriteAdTableViewCell/FavoriteAdTableViewCell.swift]: How about `statusRibbon = RibbonView(withAutoLayout: true)`?

Comment (bstien) [reply Sources/Cells/FavoriteAdTableViewCell/FavoriteAdTableViewCell.swift]: I'll see if I can do that easily for `RibbonView`👍  I have to make some changes there for it to work

**Comment by bstien** [comment]
We're also missing labels to display distance or last updated when sorting the list. These were not in the design, so I'll have to add them later.

**Comment by bstien** [comment]
And I see now that we should probably expand the touch area for the more button. It's only 24x24pts

**Pull request merged**